src/evalfunc.c

```diff
@@ -8359,9 +8359,15 @@ f_line(typval_T *argvars, typval_T *rettv)
 	{
 	    if (switch_win_noblock(&switchwin, wp, tp, TRUE) == OK)
 	    {
+		// in diff mode, prevent that the window scrolls
+		// and keep the topline
+		if (curwin->w_p_diff && switchwin.sw_curwin->w_p_diff)
```

Comment:
Why is `w_p_diff` from `sw_curwin` (i.e. the old `curwin`) also checked?

Reply:
What two are "both" buffers? If there are 3 windows (say `1000`, `1001`, `1002`) with 2 (say `1001` and `1002`) in diff mode, should the behavior of calling `line('w0', 1002)` be different depending on whether the current window is `1000` or `1001`?

Reply:
I went with the minimal approach, that changes the behaviour only if we are temporarily switching from one diff buffer to another diff buffer. And only then I think updating the topline could be skipped.

But if we are either switching from a non-diff buffer to a diff buffer (or from a diff buffer to a non-diff buffer), I think the topline should still be updated.